int32 left  # In effort
int32 right # In effort
